controlplane/kubeadm/internal/workload_cluster_conditions.go

```diff
@@ -0,0 +1,532 @@
+			continue
+		}
+
+		// Gets the list etcd members known by this member.
+		currentMembers, err := etcdClient.Members(ctx)
+		if err != nil {
+			// NB. this should never happen, given that we just received answer to the etcd calls included in forNodes;
+			// however, in case, we are considering this a signal of etcd being not stable.
+			conditions.MarkFalse(machine, controlplanev1.MachineEtcdMemberHealthyCondition, controlplanev1.EtcdMemberUnhealthyReason, clusterv1.ConditionSeverityError, "Failed get answer from the etcd member on the %s node", node.Name)
+			continue
+		}
+
+		// Check if the list of members IDs reported is the same as all other members.
+		// NOTE: the first member reporting this information is the baseline for this information.
+		if members == nil {
+			members = currentMembers
+		}
+		if !etcdutil.MemberEqual(members, currentMembers) {
+			conditions.MarkFalse(machine, controlplanev1.MachineEtcdMemberHealthyCondition, controlplanev1.EtcdMemberUnhealthyReason, clusterv1.ConditionSeverityError, "etcd member reports the cluster is composed by members %s, but all previously seen etcd members are reporting %s", etcdutil.MemberNames(currentMembers), etcdutil.MemberNames(members))
+			continue
+		}
+
+		// Check that the member reports no alarms.
+		// NB. The member for this node always exists given forNodes(node) used above
+		memberForNode := etcdutil.MemberForName(currentMembers, node.Name)
+		if len(memberForNode.Alarms) > 0 {
+			alarmList := []string{}
+			for _, alarm := range memberForNode.Alarms {
+				switch alarm {
+				case etcd.AlarmOk:
+					continue
+				default:
+					alarmList = append(alarmList, etcd.AlarmTypeName[alarm])
+				}
+			}
+			if len(alarmList) > 0 {
+				conditions.MarkFalse(machine, controlplanev1.MachineEtcdMemberHealthyCondition, controlplanev1.EtcdMemberUnhealthyReason, clusterv1.ConditionSeverityError, "Etcd member reports alarms: %s", strings.Join(alarmList, ", "))
+				continue
+			}
+		}
+
+		// Check if the member belongs to the same cluster as all other members.
+		// NOTE: the first member reporting this information is the baseline for this information.
+		if clusterID == nil {
+			clusterID = &memberForNode.ClusterID
+		}
+		if *clusterID != memberForNode.ClusterID {
+			conditions.MarkFalse(machine, controlplanev1.MachineEtcdMemberHealthyCondition, controlplanev1.EtcdMemberUnhealthyReason, clusterv1.ConditionSeverityError, "etcd member has cluster ID %d, but all previously seen etcd members have cluster ID %d", memberForNode.ClusterID, *clusterID)
+			continue
+		}
+
+		conditions.MarkTrue(machine, controlplanev1.MachineEtcdMemberHealthyCondition)
+	}
+
+	// If at lest one etcd member reported the list of etcd members, check consistency between members and machines.
+	if members != nil {
```

Comment:
Do we actually need this check? If there are no members the for loop will end right away

Reply:
I have added this after observing the following behavior live:
if there are machines, but for any reason (e.g communication errors) the list of members is empty, the first for loop will report "missing members" on all the machines, which is really alarming for a user.

So I decided to compare machines and members only when we actually know the list of members